pyflakes/checker.py

```diff
@@ -870,6 +895,33 @@ def ignore(self, node):
     # additional node types
     COMPREHENSION = KEYWORD = FORMATTEDVALUE = handleChildren
 
+    def DICT(self, node):
+        keys = node.keys
+        keys = [self.convert_to_value(key) for key in keys]
+        for key in set(keys):
+            if keys.count(key) > 1:
+                if isinstance(key, VariableKey):
+                    self.report(messages.DuplicateVariableDictionaryKey,
+                                node, key.name)
+                else:
+                    self.report(messages.DuplicateDictionaryKey, node, key)
+        self.handleChildren(node)
+
+    def convert_to_value(self, item):
+        if isinstance(item, ast.Str):
+            return item.s
+        elif isinstance(item, ast.Tuple):
+            return tuple(self.convert_to_value(i) for i in item.elts)
+        elif isinstance(item, ast.Num):
+            return item.n
+        elif isinstance(item, ast.Name):
```

Comment:
(thx, I wish there was a way to force a pr thread to be marked as done. I find github to be very confusing for in-depth code reviewing with multiple incremental revisions being made. The result seems to be that committers tend to merge early and do fixups after, creating churn and often bugs.)